backend_new/test/integration/permission.e2e-spec.ts

```diff
@@ -0,0 +1,12833 @@
+      await unitTypeFactoryAll(prisma);
+      await await prisma.translations.create({
+        data: translationFactory(),
+      });
+    });
+
+    it('should succeed for list endpoint', async () => {
+      const queryParams: ApplicationQueryParams = {
+        limit: 2,
+        page: 1,
+        order: OrderByEnum.ASC,
+        orderBy: ApplicationOrderByKeys.createdAt,
+        listingId: randomUUID(),
+      };
+      const query = stringify(queryParams as any);
+
+      await request(app.getHttpServer())
+        .get(`/applications?${query}`)
+        .expect(200);
+    });
+
+    it('should succeed for retrieve endpoint', async () => {
+      const unitTypeA = await unitTypeFactorySingle(
+        prisma,
+        UnitTypeEnum.oneBdrm,
+      );
+
+      const applicationA = await prisma.applications.create({
+        data: applicationFactory({ unitTypeId: unitTypeA.id }),
+        include: {
+          applicant: true,
+        },
+      });
+
+      await request(app.getHttpServer())
+        .get(`/applications/${applicationA.id}`)
+        .expect(200);
+    });
+
+    it('should succeed for delete endpoint & create an activity log entry', async () => {
+      const unitTypeA = await unitTypeFactorySingle(
+        prisma,
+        UnitTypeEnum.oneBdrm,
+      );
+      const jurisdiction = await prisma.jurisdictions.create({
+        data: jurisdictionFactory(`permission juris 2`),
+      });
+      const listing1 = await listingFactory(jurisdiction.id, prisma);
+      const listing1Created = await prisma.listings.create({
+        data: listing1,
+      });
+      const applicationA = await prisma.applications.create({
+        data: applicationFactory({
+          unitTypeId: unitTypeA.id,
+          listingId: listing1Created.id,
+        }),
+        include: {
+          applicant: true,
+        },
+      });
+
+      await request(app.getHttpServer())
+        .delete(`/applications/`)
+        .send({
+          id: applicationA.id,
+        })
+        .set('Cookie', cookies)
+        .expect(200);
+
+      const activityLogResult = await prisma.activityLog.findFirst({
+        where: {
+          module: 'application',
+          action: permissionActions.delete,
+          recordId: applicationA.id,
+        },
+      });
+
+      expect(activityLogResult).not.toBeNull();
+    });
+
+    it('should succeed for public create endpoint', async () => {
+      const unitTypeA = await unitTypeFactorySingle(
+        prisma,
+        UnitTypeEnum.oneBdrm,
+      );
+      const jurisdiction = await prisma.jurisdictions.create({
+        data: jurisdictionFactory(`permission juris 3`),
+      });
+      const listing1 = await listingFactory(jurisdiction.id, prisma);
+      const listing1Created = await prisma.listings.create({
+        data: listing1,
+      });
+
+      const submissionDate = new Date();
+      const exampleAddress = addressFactory() as AddressCreate;
+      const dto: ApplicationCreate = {
```

Comment:
Would be nice to see this mock data live somewhere else shared

Reply:
Hello! this comment has resulted in a drop from 12800 lines to 7400 lines
almost a 50% reduction!